crates/core/src/wit.rs

```diff
@@ -241,3 +247,64 @@ pub fn parse_wit_package_name(p: impl AsRef<str>) -> Result<WitInformationTuple>
         version,
     ))
 }
+
+#[cfg(test)]
+mod test {
+    use semver::Version;
+
+    use super::parse_wit_package_name;
+    #[test]
+    fn test_parse_wit_package_name() {
+        let (ns, packages, interfaces, func, version) =
+            parse_wit_package_name("wasi:http").expect("should have parsed'wasi:http'");
+        assert_eq!(ns, "wasi".to_string());
+        assert_eq!(packages, vec!["http".to_string()]);
+        assert_eq!(interfaces, None);
+        assert_eq!(func, None);
+        assert_eq!(version, None);
+
+        let (ns, packages, interfaces, func, version) = parse_wit_package_name("wasi:http@0.2.2")
+            .expect("should have parsed 'wasi:http@0.2.2'");
+        assert_eq!(ns, "wasi".to_string());
+        assert_eq!(packages, vec!["http".to_string()]);
+        assert_eq!(interfaces, None);
+        assert_eq!(func, None);
+        assert_eq!(version, Version::parse("0.2.2").ok());
+
+        let (ns, packages, interfaces, func, version) =
+            parse_wit_package_name("wasmcloud:bus/guest-config")
+                .expect("should have parsed 'wasmcloud:bus/guest-config'");
+        assert_eq!(ns, "wasmcloud");
+        assert_eq!(packages, vec!["bus".to_string()]);
+        assert_eq!(interfaces, Some(vec!["guest-config".to_string()]));
+        assert_eq!(func, None);
+        assert_eq!(version, None);
+
+        let (ns, packages, interfaces, func, version) =
+            parse_wit_package_name("wasmcloud:bus/guest-config.get")
+                .expect("should have parsed 'wasmcloud:bus/guest-config.get'");
+        assert_eq!(ns, "wasmcloud");
+        assert_eq!(packages, vec!["bus".to_string()]);
+        assert_eq!(interfaces, Some(vec!["guest-config".to_string()]));
+        assert_eq!(func, Some("get".to_string()));
+        assert_eq!(version, None);
+
+        let (ns, packages, interfaces, func, version) =
+            parse_wit_package_name("wasi:http/incoming-handler@0.2.0")
+                .expect("should have parsed 'wasi:http/incoming-handler@0.2.0'");
+        assert_eq!(ns, "wasi".to_string());
+        assert_eq!(packages, vec!["http".to_string()]);
+        assert_eq!(interfaces, Some(vec!["incoming-handler".to_string()]));
+        assert_eq!(func, None);
+        assert_eq!(version, Version::parse("0.2.0").ok());
+
+        let (ns, packages, interfaces, func, version) =
+            parse_wit_package_name("wasi:keyvalue/atomics.increment@0.2.0-draft")
+                .expect("should have parsed 'wasi:keyvalue/atomics.increment@0.2.0-draft'");
+        assert_eq!(ns, "wasi".to_string());
+        assert_eq!(packages, vec!["keyvalue".to_string()]);
+        assert_eq!(interfaces, Some(vec!["atomics".to_string()]));
+        assert_eq!(func, Some("increment".to_string()));
+        assert_eq!(version, Version::parse("0.2.0-draft").ok());
```

Comment:
Did you by chance miss the doctests? I think the new cases here are `wasi:http` and `wasi:http@0.2.2` so if you could put them up there that would be great

Reply:
so tl;dr why I'm adding them here is:
1. `cargo nextest`, which is what we use for CI, [does not support running doctests](https://github.com/nextest-rs/nextest/issues/16).
2. `nixify`, which is what we use for running tests, didn't have the ability to run doctests until [support for it just landed recently](https://github.com/ipetkov/crane/pull/720), but since that hasn't yet made it to `nixify` (@rvolosatovs mentioned he was looking to add it next week), I figured it would be nice to run these.

Reply:
Ah this is news to me, thanks for explaining!

Reply:
No worries, I was a little surprised to find this out myself!